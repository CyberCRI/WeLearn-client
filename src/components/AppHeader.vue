<template>
  <div class="app-header">
    <router-link to="/">
      <img alt="logo" class="logo" src="@/assets/logo.png" />
    </router-link>
  </div>
</template>

<style scoped>
.app-header {
  margin: 1rem 0 1em 1rem;
  z-index: 1;
  transition: height 0.5s;
}

@media screen and (min-width: 576px) {
  .logo {
    width: 100%;
    min-width: 4em;
    max-height: 2em;
  }
}

@media screen and (min-width: 768px) {
  .logo {
    font-size: 2rem;
  }
}

@media screen and (min-width: 992px) {
  .app-header {
    margin-left: 2rem;
    margin-bottom: 0.5rem;
  }
}
</style>
