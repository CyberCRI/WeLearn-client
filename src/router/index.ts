import { createRouter, createWebHistory, type RouteLocationNormalized } from 'vue-router';
import SearchSDG from '@/views/SearchSDG.vue';
import QandA from '@/views/QandA.vue';
import Bookmarks from '@/views/BookmarkdSources.vue';
import Tutor from '@/views/TutorPage.vue';
import LandingPage from '@/views/LandingPage.vue';
import NotFound from '@/views/NotFound.vue';

const mapEnv = {
  dev: 'dev',
  prod: 'prod'
};
const { VITE_ENVIRONMENT } = import.meta.env as 'dev' | 'prod';

const router = createRouter({
  history: createWebHistory(import.meta.env.BASE_URL),
  routes: [
    {
      path: '/',
      name: 'landing',
      component: LandingPage,
      meta: { hideTemplate: true }
    },
    {
      path: '/search',
      name: 'search-sdg',
      component: SearchSDG
    },
    {
      path: '/q-and-a',
      name: 'q-and-a',
      component: QandA
    },
    {
      path: '/bookmarks',
      name: 'bookmarks',
      component: Bookmarks
    },
    {
      ...(mapEnv[VITE_ENVIRONMENT] !== 'prod' && {
        path: '/tutor',
        name: 'tutor',
        component: Tutor
      })
    },
    { path: '/:pathMatch(.*)*', name: 'NotFound', component: NotFound }
  ]
});

function hasQueryParams(route: RouteLocationNormalized) {
  return !!Object.keys(route.query).length;
}

router.beforeEach((to, from, next) => {
  if (!hasQueryParams(to) && hasQueryParams(from)) {
    next({ name: to.name as string, query: from.query });
  } else {
    next();
  }
});

export default router;
