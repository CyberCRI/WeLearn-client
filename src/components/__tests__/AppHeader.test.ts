import { describe, it, expect } from 'vitest';

import router from '@/router';

import { shallowMount } from '@vue/test-utils';
import AppHeader from '../AppHeader.vue';

const mountOptions = {
  global: {
    plugins: [router]
  }
};

describe('AppHeader', () => {
  it('renders properly', () => {
    const wrapper = shallowMount(AppHeader, mountOptions);
    expect(wrapper.element).toMatchSnapshot();
  });
});
