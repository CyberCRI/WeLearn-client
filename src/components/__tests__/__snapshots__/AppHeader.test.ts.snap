// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppHeader > renders properly 1`] = `
<div
  class="app-header"
  data-v-48a13218=""
>
  <router-link-stub
    ariacurrentvalue="page"
    custom="false"
    data-v-48a13218=""
    replace="false"
    to="/"
  />
</div>
`;
